sklearn/neighbors/_base.py

```diff
@@ -784,9 +784,15 @@ class from an array representing our data set and ask who's
 
         n_samples_fit = self.n_samples_fit_
         if n_neighbors > n_samples_fit:
+            if query_is_train:
+                n_neighbors -= 1
+                inequality_str = "n_neighbors < n_samples_fit"
+            else:
+                inequality_str = "n_neighbors <= n_samples_fit"
```

Comment:
Could you add a test case for this scenario as well?

Reply:
@Kshitij68 I'm not sure how to reproduce this scenario/hit this case because the `query_is_train` flag is invoked when calling `fit` or `fit_predict` and that is a necessary prerequisite before the `predict` or other test time methods are invoked.

Reply:
You can call `kneighbors()` with and without providing `X` which effects the value of `is_query_train`. This should allow you to make a test for each of the two cases.